whichone/static/js/data.js

```diff
@@ -1,103 +1,68 @@
-function localDataFound() {
-  // get data from local storage instead of the web server
-  let data = localStorage.getItem(spotify_id);
-  const EXPIRATION_TIME = 604800000; // 1 week (milliseconds)
+function isLocalDataAvailable() {
+  return localStorage.getItem(spotify_id) != null;
+}
 
-  if (data != null) {
-    user = JSON.parse(data);
-    // check if modes are present
-    let resetModes = false;
-    if (user.modes != null) {
-      if (
-        user.modes.dance == null ||
-        user.modes.upbeat == null ||
-        user.modes.duration == null
-      ) {
-        resetModes = true;
-      }
-    } else {
-      resetModes = true;
-    }
+function isLocalDataExpired(storedTimestamp, currentTimestamp, expirationTime) {
+  // check if their spotify data has expired
+  return currentTimestamp - storedTimestamp > expirationTime;
+}
+
+const isBoolean = (val) => "boolean" === typeof val;
+
+function getLocalData(user_id) {
+  // get data from local storage instead of the web server
+  let data = localStorage.getItem(user_id);
+  let local = JSON.parse(data);
 
-    if (resetModes) {
-      user.modes = {
+  // check if modes are present and valid
+  if (local.modes != null) {
+    if (
+      !isBoolean(local.modes.dance) ||
+      !isBoolean(local.modes.upbeat) ||
+      !isBoolean(local.modes.duration)
+    ) {
+      local.modes = {
         dance: true,
         upbeat: true,
         duration: true,
       };
     }
+  }
 
-    if (user.high_score != null) {
-      if (Number.isNaN(user.high_score)) {
-        user.high_score = 0;
-      } else {
-        document.getElementById("high_score-m").textContent = user.high_score;
-        document.getElementById("high_score-d").textContent = user.high_score;
-      }
-    } else {
-      user.high_score = 0;
-    }
-
-    // check if their spotify data is outdated
-    if (user.expire != null) {
-      // If the expiration time has passed, we need to get new data instead of using the local storage
-      if (new Date().getTime() - new Date(user.expire) > EXPIRATION_TIME) {
-        return false;
-      }
-    } else {
-      return false;
-    }
-
-    return true;
-  } else {
-    return false;
+  if (local.high_score == null || Number.isNaN(local.high_score)) {
+    local.high_score = 0;
   }
+  return local;
 }
 
-async function getAudioFeatures() {
-  let trackIds = [];
-  if (user.top_tracks != null && user.top_tracks.length > 0) {
-    user.top_tracks.forEach((track) => {
-      trackIds.push(track.id);
-    });
-
+async function post_request(url, track_ids) {
```

Comment:
No need to be async

Reply:
@mcgill-a @athompsonScottLogic You need async only if you are awaiting within the function. https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/async_function